frontend_tests/casper_tests/11-mention.js

```diff
@@ -16,39 +18,53 @@ casper.then(function () {
 });
 common.select_item_via_typeahead('#new_message_content', '@all', 'all');
 
-casper.waitForText("Are you sure you want to mention all", function () {
-    casper.test.info('Warning message appears when mentioning @all');
-    casper.test.assertSelectorHasText('.compose-all-everyone', 'Are you sure you want to mention all');
-});
-
-casper.then( function () {
+casper.then(function () {
     common.turn_off_press_enter_to_send();
     casper.test.info('Click Send Button');
     casper.click('#compose-send-button');
 });
-casper.waitForText("Please remove @all", function () {
-    casper.test.info('Error message appears when attempting to send a message without acknowledging the @all mention warning');
-    casper.test.assertSelectorHasText('#error-msg', "Please remove @all / @everyone or acknowledge that you will be spamming everyone!");
-});
 
-casper.waitForSelector('.compose-all-everyone-confirm', function () {
-    casper.click('.compose-all-everyone-confirm');
-}, function () {
-    casper.test.error('Could not click confirm button.');
+casper.then(function () {
+    casper.test.info("Checking for all everyone warning");
+    var stream_size = this.evaluate(function () {
+        return stream_data.get_sub('Verona').subscribers.num_items();
+    });
+    var threshold = this.evaluate(function () {
+        return compose.all_everyone_warn_threshold;
+    });
+    is_warning = stream_size > threshold;
 });
 
-casper.waitWhileVisible('.compose-all-everyone-confirm', function () {
-    casper.test.info('Check that error messages are gone.');
-    casper.test.assertNotVisible('.compose-all-everyone-msg');
-    casper.test.assertNotVisible('#send-status');
-});
+casper.then(function () {
+    if (is_warning) {
+        casper.test.info('Warning appeared');
+        casper.waitForText("Are you sure you want to message all", function () {
+            casper.test.info('Warning message appears when mentioning @all');
+            casper.test.assertSelectorHasText('.compose-all-everyone-msg', 'Are you sure you want to message all');
+        });
 
-casper.then( function () {
-    casper.test.info('Click Send Button');
-    casper.click('#compose-send-button');
+        casper.waitForSelector('.compose-all-everyone-confirm', function () {
+            casper.click('.compose-all-everyone-confirm');
+        }, function () {
+            casper.test.error('Could not click confirm button.');
+        });
+
+        casper.waitWhileVisible('.compose-all-everyone-confirm', function () {
+            casper.test.info('Check that error messages are gone.');
+            casper.test.assertNotVisible('.compose-all-everyone-msg');
+            casper.test.assertNotVisible('#send-status');
+        });
+
+        casper.then( function () {
```

Comment:
`casper.then` shouldn't be nested; you should make this a separate `casper.then` call.